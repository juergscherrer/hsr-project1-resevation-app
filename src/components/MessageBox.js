import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';

class MessageBox extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <React.Fragment>
        <Snackbar
          anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
          open={this.props.open}
          onClose={this.props.onClose}
          ContentProps={{
            'aria-describedby': 'message-id'
          }}
          message={<span id="message-id">{this.props.message || ''}</span>}
          autoHideDuration={6000}
        />
      </React.Fragment>
    );
  }
}

export default MessageBox;
