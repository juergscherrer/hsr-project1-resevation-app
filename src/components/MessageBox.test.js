import React from 'react';
import { shallow } from 'enzyme';
import MessageBox from './MessageBox';

let mockDeleteMessage = () => {
  return true;
};

describe('<MessageBox />', () => {
  it('should be defined', () => {
    expect(MessageBox).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <MessageBox
        open={true}
        message="Test Message"
        onClose={mockDeleteMessage}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a message prop', () => {
    const wrapper = shallow(
      <MessageBox
        open={true}
        message="Test Message"
        onClose={mockDeleteMessage}
      />
    );

    expect(typeof wrapper.instance().props.message).toBe('string');
    expect(wrapper.instance().props.message).toEqual('Test Message');
  });

  it('should have a open prop', () => {
    const wrapper = shallow(
      <MessageBox
        open={true}
        message="Test Message"
        onClose={mockDeleteMessage}
      />
    );

    expect(typeof wrapper.instance().props.open).toBe('boolean');
    expect(wrapper.instance().props.open).toBeTruthy();
  });

  it('should call handleAnswer and return true', () => {
    const wrapper = shallow(
      <MessageBox
        open={true}
        message="Test Message"
        onClose={mockDeleteMessage}
      />
    );

    expect(typeof wrapper.instance().props.deleteMessage).toBeTruthy();
  });
});
