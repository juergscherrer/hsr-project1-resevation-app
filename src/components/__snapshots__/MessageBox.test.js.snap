// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBox /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageBox
    message="Test Message"
    onClose={[Function]}
    open={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WithStyles(Snackbar)
        ContentProps={
          Object {
            "aria-describedby": "message-id",
          }
        }
        anchorOrigin={
          Object {
            "horizontal": "left",
            "vertical": "bottom",
          }
        }
        autoHideDuration={6000}
        message={
          <span
            id="message-id"
          >
            Test Message
          </span>
        }
        onClose={[Function]}
        open={true}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ContentProps": Object {
          "aria-describedby": "message-id",
        },
        "anchorOrigin": Object {
          "horizontal": "left",
          "vertical": "bottom",
        },
        "autoHideDuration": 6000,
        "message": <span
          id="message-id"
        >
          Test Message
        </span>,
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(Snackbar)
          ContentProps={
            Object {
              "aria-describedby": "message-id",
            }
          }
          anchorOrigin={
            Object {
              "horizontal": "left",
              "vertical": "bottom",
            }
          }
          autoHideDuration={6000}
          message={
            <span
              id="message-id"
            >
              Test Message
            </span>
          }
          onClose={[Function]}
          open={true}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ContentProps": Object {
            "aria-describedby": "message-id",
          },
          "anchorOrigin": Object {
            "horizontal": "left",
            "vertical": "bottom",
          },
          "autoHideDuration": 6000,
          "message": <span
            id="message-id"
          >
            Test Message
          </span>,
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
